src/lib/utils.ts

```diff
@@ -282,23 +283,55 @@ export function encodeKey(
     case 'mappingwithgrouping':
     case 'singleton':
     case 'mapping':
+      if (schema.valueType.includes('[CompactBytesArray]')) {
+        const valueType = schema.valueType.replace('[CompactBytesArray]', '');
+        const valueContent = schema.valueContent.replace(
+          '[CompactBytesArray]',
+          '',
+        );
+
+        if (
+          Array.isArray(value) &&
+          Array.isArray(value[0]) &&
+          valueType[0] === '(' &&
+          valueType[valueType.length - 1] === ')'
+        ) {
+          const encodedTuples = value.map((element) => {
+            return encodeTupleKeyValue(valueContent, valueType, element);
+          });
+          return encodeValueType('bytes[CompactBytesArray]', encodedTuples);
+        }
+        if (Array.isArray(value) && !Array.isArray(value[0]))
+          return encodeValueType(schema.valueType, value as string[]);
```

Comment:
could we add some validation somewhere to ensure value is a string array at this point? e.g. here it could be an array of numbers